micros/modules/networking.nix

```
@@ -107,12 +109,174 @@ in {
           The set of NTP servers from which to synchronise.
         '';
       };
-    };
-  };
 
-  config = {
-    environment.etc.hostname = mkIf (config.networking.hostName != "") {
-      text = config.networking.hostName + "\n";
+      interfaces = mkOption {
+        default = {};
+        type = types.attrsOf (types.submodule {
+          options = {
+            method = mkOption {
+              type = types.enum ["dhcp" "wifi" "static"];
+              default = "dhcp";
+              description = "Specifies the network configuration method for the interface.";
+            };
```

What If someone wants to use static addressing with a wifi connection?
 I'd suggest seperating connection media from the "where do I get my IP address from" part:
 ```suggestion
            ipAcquisitionMethod = mkOption {
              type = types.enum ["dhcp" "static"];
              default = "dhcp";
              description = "Specifies how to acquire an IP address for the interface.";
            };
            connectionType = mkOption {
              type = types.enum ["wifi" "ethernet"];
              default = "ethernet";
              description = "Specifies the connection type used by the interface.";
            };
```

---

wifi is a problem I'd like to think about later. Good idea, but networking is too annoying to work on